platform/disk/parted_partitioner.go

```diff
@@ -370,3 +370,86 @@ func (p partedPartitioner) createMapperPartition(devicePath string) error {
 	detectPartitionRetryStrategy := NewPartitionStrategy(detectPartitionRetryable, p.timeService, p.logger)
 	return detectPartitionRetryStrategy.Try()
 }
+
+func (p partedPartitioner) PartitionsNeedResize(devicePath string, partitionsToMatch []Partition) (needsResize bool, err error) {
+	existingPartitions, _, err := p.GetPartitions(devicePath)
+	if err != nil {
+		return false, bosherr.WrapError(err, "Failed to get existing partitions")
+	}
+	if len(existingPartitions) < len(partitionsToMatch) {
+		return false, nil
+	}
+
+	remainingDiskSpace, err := p.GetDeviceSizeInBytes(devicePath)
+	if err != nil {
+		return false, bosherr.WrapError(err, "Failed to get device size")
+	}
+
+	for index, partitionToMatch := range partitionsToMatch {
+		if index == len(partitionsToMatch)-1 {
+			partitionToMatch.SizeInBytes = remainingDiskSpace
+		}
+
+		existingPartition := existingPartitions[index]
+		switch {
+		case existingPartition.Type != partitionToMatch.Type:
+			return false, nil
+		case significantlySmallerThan(existingPartition.SizeInBytes, partitionToMatch.SizeInBytes, ConvertFromMbToBytes(deltaSize)):
+			return true, nil
+		}
+
+		remainingDiskSpace = remainingDiskSpace - partitionToMatch.SizeInBytes
+	}
+
+	return false, nil
+}
+
+func (p partedPartitioner) ResizePartitions(devicePath string, partitionsToMatch []Partition) (err error) {
```

Comment:
This function does not need `partitionsToMatch` it is only used to check the SizeInBytes which is never set by the caller anyway.

Reply:
All right, I'll rework the signature of this method and simplify the algorithm.